### PR TITLE
Add || operator for Boolean objects.

### PR DESCRIPTION
**Pull request by ummarikar**
Add || operator for Boolean objects along with tests - Boolean || Block tests (including tests which ensure that the Block is not evaluated when the Boolean is true and is evaluated when the Boolean is false) and Boolean || Boolean tests.

### CODE REVIEW

File: lang_tests/bool4.som [hunk 1]
```diff
@@ -7,7 +7,7 @@ VM:
     false
     false
     false
-    true
+    true
```
Comment (ltratt) [inline lang_tests/bool4.som]: Trailing space.

**Comment by ltratt** [comment]
Other than the trailing space this looks good.

**Comment by ummarikar** [comment]
Done!

**Comment by ltratt** [comment]
Please squash.

**Comment by ummarikar** [comment]
Squashed.

**Pull request merged**